### PR TITLE
Fix error misalignment for Quick order list

### LINKED ISSUE #2886: Quick order list: Error messages that wrap push all other elements in the row down (vertically centered) causing misalignment
When error messages appear in a row on the Quick order list, all other elements in the row get pushed down. This might be because of the vertically centering of elements which was done in previous PR's to fix alignment issues.

- [x] Can we have the row keep all elements aligned but have the error messaging at the bottom below the qty picker? 
- [x] The error messaging should also allow for the row to grow taller without elements getting pushed down

@melissaperreault's video: https://screenshot.click/17-48-6lu4r-k1m58.mp4
Expected behaviour video: https://screenshot.click/Figma-expected-behaviour-error-messages-below.mp4

https://screenshot.click/Current-PR-Error-messaging.png

Image

### PR DESCRIPTION
**Pull request by eugenekasimov**
### PR Summary: 

This PR fixes error misalignment for Quick order list

### Why are these changes introduced?

Fixes #2886.

### What approach did you take?

Add extra <tr> to table and put error message there. Hide this `<tr>` for mobile. Hide original `<div>` with error message for desktop and make it visible for mobile only. 

### Testing steps/scenarios

- [ ] Step 1

### Demo links

- [Editor](https://admin.shopify.com/store/os2-demo/themes/140323160086/editor)

### Checklist
- [ ] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: assets/quick-order-list.js [hunk 1]
```diff
@@ -334,8 +334,13 @@ class QuickOrderList extends HTMLElement {
   }
```
Comment (sofiamatulis) [inline assets/quick-order-list.js:334]: Looks good 👌 I think the spacing is a bit far now. But ill let @edmund-teh and @nicklepine take a look! https://screenshot.click/21-41-ihm3k-fnjbu.png

Comment (nicklepine) [reply assets/quick-order-list.js:334]: Thanks, yes is there any way to reduce some of the vertical spacing to be closer to what it used to be
reference pre-this pr: https://screenshot.click/21-11-f021j-blp5l.png

**Comment by edmund-teh** [comment]
> Works well!
> 
> We'll need to reduce the top padding of the error message on desktop to ensure there's no regression and make sure to test with input border/shadow global settings.
> 
> > We had some feedback that the spacing between the error message and its parent UI could be improved. Can we make things a bit tighter here? The current implementation makes it hard to distinguish if it’s related to the input above or below.
> > https://screenshot.click/Figma-Error-tighter-spacing.png
> > [Figma file here](https://www.figma.com/file/ov2FHR3uvfVzKqbVV33IPz/Variant-list-on-PDP?type=design&node-id=4592-102380&mode=design&t=pwynUUCQnntJmpr7-4)
> 
> Originally posted here: [#2627 (comment)](https://github.com/Shopify/dawn/pull/2627#issuecomment-1621047006)


Good eye @KaichenWang and thanks for referencing that previous [comment](https://github.com/Shopify/dawn/pull/2627#issuecomment-1621047006)! 

I've received feedback several times on how having more spacing there makes it hard to tell if the error messaging is for the variant row above or below. Let's bring things in tighter. Thanks @eugenekasimov!

**Review by KaichenWang — APPROVED** [review]

**Review by edmund-teh — APPROVED** [review]
Just checked this on mobile, tablet, and desktop! Looks great @eugenekasimov— thank you!

**Review by sofiamatulis — APPROVED** [review]

**Pull request merged**